client-cli/src/command/transaction_command.rs

```diff
@@ -204,17 +213,120 @@ fn new_unbond_transaction<N: NetworkOpsClient>(
 }
 
-fn new_deposit_transaction<N: NetworkOpsClient>(
+fn new_deposit_transaction<T: WalletClient, N: NetworkOpsClient>(
+    wallet_client: &T,
     network_ops_client: &N,
     name: &str,
     enckey: &SecKey,
 ) -> Result<TxAux> {
     let attributes = StakedStateOpAttributes::new(get_network_id());
     let inputs = ask_inputs()?;
     let to_address = ask_staking_address()?;
+    if !wallet_client.has_unspent_transactions(name, enckey, &inputs)? {
+        return Err(Error::new(
+            ErrorKind::InvalidInput,
+            "Given transaction inputs are not present in unspent transactions (synchronizing your wallet may help)",
+        ));
+    }
+    let transactions = inputs
+        .into_iter()
+        .map(|txo_pointer| {
+            let output = wallet_client.output(name, enckey, &txo_pointer)?;
+            Ok((txo_pointer, output))
+        })
+        .collect::<Result<Vec<(TxoPointer, TxOut)>>>()?;
+    network_ops_client.create_deposit_bonded_stake_transaction(
+        name,
+        enckey,
+        transactions,
+        to_address,
+        attributes,
+    )
+}
 
-    network_ops_client
-        .create_deposit_bonded_stake_transaction(name, enckey, inputs, to_address, attributes)
+fn new_amount_deposit_transaction<T: WalletClient, N: NetworkOpsClient>(
+    wallet_client: &T,
+    network_ops_client: &N,
+    name: &str,
+    enckey: &SecKey,
+) -> Result<()> {
+    let to_staking_address = ask_staking_address()?;
+    let attr = StakedStateOpAttributes::new(get_network_id());
+    ask("Enter deposit amount (in CRO): ");
+    let amount_str = text().chain(|| (ErrorKind::IoError, "Unable to read amount"))?;
+    let amount = coin_from_str(&amount_str)?;
+    let fee = network_ops_client.calculate_deposit_fee()?;
+    let total_amount = (amount + fee).chain(|| (ErrorKind::InvalidInput, "invalid amount"))?;
+    success(&format!(
+        "create a transfer transaction to make a UTXO with {} amount(fee is {})",
+        total_amount, fee
+    ));
+    let to_transfer_address = wallet_client.new_transfer_address(name, enckey)?;
+    let tx_id = wallet_client.send_to_address(
+        name,
+        enckey,
+        total_amount,
+        to_transfer_address,
+        vec![],
+        get_network_id(),
+    )?;
+
+    success("broadcasting transfer transaction");
+    let block_height = wallet_client.get_current_block_height()?;
+    loop {
+        // query tx_id from tendermint
+        print!(".");
+        io::stdout().flush().unwrap();
+        if !wallet_client.check_tx_confirmed(tx_id) {
+            std::thread::sleep(Duration::from_secs(1));
+            let current_block_height = wallet_client.get_current_block_height()?;
+            if current_block_height - block_height >= 50 {
+                return Err(Error::new(
+                    ErrorKind::InternalError,
+                    "broadcast transaction timeout",
+                ));
+            }
+            continue;
+        }
+        break;
+    }
```

Comment:
I haven't check if in `Tendermint 0.32`, there's a RPC call `broadcast_tx_commit` that will wait until tx is committed... if there is, it'll be simpler

Reply:
I don't think we should use it even if it is present. Tendermint documentation says that it should only be used for development and testing.

Reply:
ok, I didn't see that documentation